docs/testing/ponycheck.md

```diff
@@ -0,0 +1,102 @@
+    Generators.ascii()
+
+  fun property(arg1: String, ph: PropertyHelper) =>
+    ph.assert_eq[String](arg1, arg1)
+```
+
+A `Property1` needs a name for identification in test output. We created a `Generator` by using one of the many convenience factory methods and combinators defined in the [`Generators`](https://stdlib.ponylang.io/pony_check-Generators) primitive and we used [`PropertyHelper`](https://stdlib.ponylang.io/pony_check-PropertyHelper) to assert on a condition that should hold for all samples
+
+Below are two classic list reverse properties from the QuickCheck paper adapted to Pony arrays:
+
+```pony
+use "pony_check"
+use "collections"
+
+class _ListReverseProperty is Property1[Array[USize]]
+  fun name(): String => "list/reverse"
+
+  fun gen(): Generator[Array[USize]] =>
+    Generators.seq_of[USize, Array[USize]](Generators.usize())
+
+  fun property(arg1: Array[USize], ph: PropertyHelper) =>
+    ph.assert_array_eq[USize](arg1, arg1.reverse().reverse())
+
+class _ListReverseOneProperty is Property1[Array[USize]]
+  fun name(): String => "list/reverse/one"
+
+  fun gen(): Generator[Array[USize]] =>
+    Generators.seq_of[USize, Array[USize]](Generators.usize() where min=1,max=1)
+
+  fun property(arg1: Array[USize], ph: PropertyHelper) =>
+    ph.assert_array_eq[USize](arg1, arg1.reverse())
+```
+
+## Integration with PonyTest
+
+PonyCheck properties need to be executed. The test runner for PonyCheck is [PonyTest](https://stdlib.ponylang.io/pony_test--index). To integrate [`Property1`](https://stdlib.ponylang.io/pony_check-Property1) into [PonyTest](https://stdlib.ponylang.io/pony_test--index), `Property1` needs to be wrapped inside a [`Property1UnitTest`](hhttps://stdlib.ponylang.io/pony_check-Property1UnitTest) and passed to the PonyTest `apply` method as a regular PonyTest [`UnitTest`](https://stdlib.ponylang.io/pony_test-UnitTest):
+
+```pony
+use "pony_test"
+use "pony_check"
+
+actor Main is TestList
+  new create(env: Env) =>
+    PonyTest(env, this)
+
+  fun tag tests(test: PonyTest) =>
+    test(Property1UnitTest[String](_MyFirstProperty))
+```
+
+It is also possible to integrate any number of properties directly into one
+[`UnitTest`](https://stdlib.ponylang.io/pony_test-UnitTest) using the [`PonyCheck.forAll`](https://stdlib.ponylang.io/pony_check-PonyCheck) convenience function:
+
+```pony
+class _ListReverseProperties is UnitTest
+  fun name(): String => "list/properties"
+
+  fun apply(h: TestHelper) ? =>
+    let gen1 = Generators.seq_of[USize, Array[USize]](Generators.usize())
+    PonyCheck.forAll[Array[USize]](gen1, h)({
+      (arg1: Array[USize], ph: PropertyHelper) =>
+        ph.assert_array_eq[USize](arg1, arg1.reverse().reverse())
+    })
+    let gen2 = Generators.seq_of[USize, Array[USize]](1, Generators.usize())
+    PonyCheck.forAll[Array[USize]](gen2, h)({
+      (arg1: Array[USize], ph: PropertyHelper) =>
+        ph.assert_array_eq[USize](arg1, arg1.reverse())
+    })
+```
```

Comment:
Is it supposed to be `PonyCheck.for_all` as in the current API of ponylang/ponycheck or has this been changed to `PonyCheck.forAll` per style guide changes?

Reply:
I didn't that issue. Can you update here and update the PR on ponyc including noting the change in the release notes?

Reply:
So snake case is used for functions so forAll would be incorrect.

Reply:
It looks like that is only in the tutorial here, so I will update here.

Reply:
Perfect! Thank you!